### PR TITLE
[Tests-Only] Expand API tests for getUser, removeSubAdmin and resetUserPassword

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
expands apiProvisioning tests for ocs v1 and v2 in:
- getUser
- removeSubAdmin
- resetUserPassword

## Related Issue
- part of https://github.com/owncloud/core/issues/31533

## How Has This Been Tested?
- test environment: drone :robot: 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiProvisioning-v2/removeSubAdmin.feature [hunk 1]
```diff
@@ -31,6 +31,21 @@ Feature: remove subadmin
...
+    Given these users have been created with default attributes and skeleton files:
+      | username         |
+      | subadmin         |
+      | another-subadmin |
+    And group "new-group-1" has been created
+    And group "new-group-2" has been created
+    And user "subadmin" has been made a subadmin of group "new-group-1"
+    And user "another-subadmin" has been made a subadmin of group "new-group-2"
+    When user "subadmin" removes user "another-subadmin" from being a subadmin of group "new-group-2" using the provisioning API
+    Then the OCS status code should be "997"
```
Comment (Talank) [inline tests/acceptance/features/apiProvisioning-v2/removeSubAdmin.feature:45]: What is the expected behavior here? Maybe it is better to mention that in the comment :thinking:

File: tests/acceptance/features/apiProvisioning-v2/resetUserPassword.feature [hunk 2]
```diff
@@ -53,6 +64,37 @@ Feature: reset user password
...
+  Scenario: subadmin should not be able to reset the password of another subadmin of same group
+    Given these users have been created with skeleton files:
+      | username         | password   | displayname | email                      |
+      | another-subadmin | %regular%  | New user    | another.subadmin@oc.com.np |
+      | subadmin         | %subadmin% | Sub Admin   | sub.admin@oc.com.np        |
+    And group "new-group" has been created
+    And user "another-subadmin" has been made a subadmin of group "new-group"
+    And user "subadmin" has been made a subadmin of group "new-group"
+    When user "subadmin" tries to reset the password of user "another-subadmin" to "%alt1%" using the provisioning API
+    # Then the OCS status code should be "401"
```
Comment (dpakach) [inline tests/acceptance/features/apiProvisioning-v2/resetUserPassword.feature:92]: I think the ocs status 997 is correct , please remove this line

File: tests/acceptance/features/apiProvisioning-v2/removeSubAdmin.feature [hunk 3]
```diff
@@ -31,6 +31,21 @@ Feature: remove subadmin
     And the HTTP status code should be "401"
     And user "another-subadmin" should be a subadmin of group "brand-new-group"
 
+  @issue-31276
+  Scenario: subadmin tries to remove subadmin of another group
```
Comment (swoichha) [inline tests/acceptance/features/apiProvisioning-v2/removeSubAdmin.feature:35]: I agree with @Talank. But if it is the actual expected behavior(sub-admin of one group cannot delete sub-admin of another group) then you can change the Scenario to something like this 
```suggestion
  Scenario: subadmin should not be able to remove subadmin of another group
```

File: tests/acceptance/features/apiProvisioning-v1/getUser.feature [hunk 4]
```diff
@@ -78,6 +89,34 @@ Feature: get user
...
+    And the display name returned by the API should be "Regular User"
+    And the quota definition returned by the API should be "default"
+
+  Scenario: a subadmin tries to get information of another subadmin of same group
+    Given these users have been created with default attributes and skeleton files:
+      | username         |
+      | subadmin         |
+      | another-subadmin |
+    And group "brand-new-group" has been created
+    And user "another-subadmin" has been made a subadmin of group "brand-new-group"
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/acceptance/features/apiProvisioning-v1/getUser.feature:113]: This step passed in local run
We can make user subadmin of any group without adding them in that group.
(may be this behavior is a bug if the step should fail)

File: tests/acceptance/features/apiProvisioning-v2/resetUserPassword.feature (hunk 2 above)
Comment (saw-jan) [reply tests/acceptance/features/apiProvisioning-v2/resetUserPassword.feature:92]: that issue tag was a mistake
it must be: https://github.com/owncloud/core/issues/31276
shouldn't the ocs v2 API give `401`  incase of unauthorized as in HTTP?
If `997` is the correct (not a buggy) status code then it is better to be removed.

File: tests/acceptance/features/apiProvisioning-v2/removeSubAdmin.feature (hunk 3 above)
Comment (saw-jan) [reply tests/acceptance/features/apiProvisioning-v2/removeSubAdmin.feature:35]: I was just trying to continue the flow each feature file :smile:

File: tests/acceptance/features/apiProvisioning-v2/resetUserPassword.feature (hunk 2 above)
Comment (saw-jan) [reply tests/acceptance/features/apiProvisioning-v2/resetUserPassword.feature:92]: `997` is considered correct, better to remove the line

File: tests/acceptance/features/apiProvisioning-v2/removeSubAdmin.feature (hunk 1 above)
Comment (saw-jan) [reply tests/acceptance/features/apiProvisioning-v2/removeSubAdmin.feature:44]: this is the correct behavior, need to remove the issue tag

File: tests/acceptance/features/apiProvisioning-v2/getUsers.feature [hunk 5]
```diff
@@ -40,13 +40,13 @@ Feature: get users
...
-  Scenario: normal user tries to get other users
+  @issue-38332
+  Scenario: normal user gets all users
     Given these users have been created with default attributes and skeleton files:
       | username         |
       | brand-new-user   |
       | another-new-user |
     When user "brand-new-user" gets the list of all users using the provisioning API
-    Then the OCS status code should be "401"
+    Then the OCS status code should be "997"
```
Comment (dpakach) [inline tests/acceptance/features/apiProvisioning-v2/getUsers.feature:50]: This may make this test fail on on ocis

**Review by dpakach — COMMENTED** [review]
Everything looks good but you have added tests in the middle of the file which messes up the line numbers of other existing tests. That may result in many failures in ocis, so please test this branch with ocis as well.

File: tests/acceptance/features/apiProvisioning-v2/getUsers.feature (hunk 5 above)
Comment (saw-jan) [reply tests/acceptance/features/apiProvisioning-v2/getUsers.feature:50]: felt better to move changes to the end of the file

**Review by Talank — COMMENTED** [review]
LGTM

**Review by individual-it — APPROVED** [review]

**Pull request merged**